server/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java

```diff
@@ -1134,4 +1134,44 @@ public void test_update_continues_on_error() {
         execute("update test set a = a + 98");
         assertThat(response).hasRowCount(1);
     }
+
+    @Test
+    public void test_update_fail_fast() {
+        execute("create table test (a int CHECK (a < 100))");
+        execute("insert into test (a) values (1), (2), (3)");
+        execute("refresh table test");
+
+        try (var session = sqlExecutor.newSession()) {
+            session.sessionSettings().dmlFailFast(true);
+            assertSQLError(() -> execute("update test set a = a + 98", session))
+                .hasPGError(INTERNAL_ERROR)
+                .hasHTTPError(BAD_REQUEST, 4000)
+                .hasMessageContaining("Failed CONSTRAINT");
+        }
+    }
+
+    @Test
+    public void test_bulk_update_by_id_fail_fast() throws Exception {
```

Comment:
`UpdateById` must also taken the new session setting into account. Currently, the operation won't stop, but just the first error is returned.
Please adjust the test, so that it would write a valid record after the failure and then ensure that his record was never written. Same for the UpdateByQuery test.